### PR TITLE
Check for whether the running script is updated

### PR DESCRIPTION
**Pull request by sloanebernstein**
Allow the elevate-cpanel script to notify the administrator if a newer
version of the script is available. Also allow the admin to skip this
blocker if desired.

Implements #25.

By submitting pull requests to this repo, I agree to the Contributor License Agreement which can be found at: https://github.com/cpanel/elevate/blob/main/docs/cPanel-CLA.pdf

### CODE REVIEW

File: elevate-cpanel [hunk 1]
```diff
@@ -2756,6 +2759,64 @@ sub _blocker_ea4_profile( $self ) {
     EOS
 }
 
+sub _blocker_script_updated ($self) {
+
+    my ( $should_block, $blocker_text ) = ( undef, undef );
+
+    if ( !$self->getopt('skip-elevate-version-check') ) {
```
Comment (toddr) [inline elevate-cpanel:2766]: can't you just return here?

File: elevate-cpanel [hunk 2]
```diff
@@ -2756,6 +2759,64 @@ sub _blocker_ea4_profile( $self ) {
...
+
+        my ( $latest_checksum, $self_checksum ) = ( _latest_checksum(), _self_checksum() );
+        if ( !defined $latest_checksum ) {
+            $should_block = 1;
+            $blocker_text = "The latest version of elevate-cpanel could not be fetched for comparison.";
+        }
+        elsif ( !defined $self_checksum ) {
+            $should_block = 1;
+            $blocker_text = "This script is not installed at the expected location /scripts/elevate-cpanel.";
```
Comment (toddr) [inline elevate-cpanel:2775]: Should be unreachable code as we do this check already earlier. Fine to leave.

File: elevate-cpanel (hunk 1 above)
Comment (sloanebernstein) [reply elevate-cpanel:2766]: This is the legacy of previous revisions of this logic.

File: elevate-cpanel [hunk 3]
```diff
@@ -2756,6 +2759,64 @@ sub _blocker_ea4_profile( $self ) {
...
+        }
+        else {
+            $should_block = $latest_checksum eq $self_checksum ? 0 : 1;
+            $blocker_text = <<~EOS if $should_block;
+            This script does not appear to be the newest available release.
+            Please download the latest version from the following URL and try again:
+
+            https://raw.githubusercontent.com/cpanel/elevate/release/elevate-cpanel
```
[the remark this answers is not in the document]
Comment (sloanebernstein) [reply elevate-cpanel:2783]: A self-update is the logical conclusion of needing to fetch the latest release at all. We should handle that in another issue/PR.

**Pull request merged**